bumble/pandora/security.py

```diff
@@ -294,23 +294,38 @@ async def Secure(
             try:
                 self.log.debug('Pair...')
 
+                security_result = asyncio.get_running_loop().create_future()
+
+                @connection.on('pairing')
+                def on_pairing(*_: Any) -> None:
+                    security_result.set_result('success')
+
+                @connection.on('pairing_failure')
+                def on_pairing_failure(*_: Any) -> None:
+                    security_result.set_result('pairing_failure')
+
+                @connection.on('disconnection')
+                def on_disconnection(*_: Any) -> None:
+                    security_result.set_result('connection_died')
```

Comment:
I test locally with avatar and everything work fine :tada: apart for all the `security_result.set_result()` calls that I needed to wrap to avoid illegal state exceptions (it only appended on disconnect tho):
```suggestion
                    if not security_result.done():
                        security_result.set_result('connection_died')
```

Reply:
@uael maybe the event group utility in PR #292 could help ensure that only one of the event handlers is called, so only one `set_result` is invoked.

Reply:
Cool!